cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -150,246 +154,253 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
+            }
         }
     }
 
     private void scheduleBanners() {
-		if (CleverPush.getInstance(activity).areAppBannersDisabled()) {
-			pendingBanners.addAll(popups);
-			popups.removeAll(pendingBanners);
-			return;
-		}
+        if (CleverPush.getInstance(activity).areAppBannersDisabled()) {
+            pendingBanners.addAll(popups);
+            popups.removeAll(pendingBanners);
+            return;
+        }
 
         Date now = new Date();
         for (AppBannerPopup bannerPopup : popups) {
             Banner banner = bannerPopup.getData();
 
             if (banner.isScheduled()) {
-            	continue;
-			}
+                continue;
+            }
 
             banner.setScheduled();
 
             if (banner.getStartAt().before(now)) {
-				if (banner.getDelaySeconds() > 0) {
-					handler.postDelayed(() -> showBanner(bannerPopup), 1000 * banner.getDelaySeconds());
-				} else {
-					handler.post(() -> showBanner(bannerPopup));
-				}
+                if (banner.getDelaySeconds() > 0) {
+                    handler.postDelayed(() -> showBanner(bannerPopup), 1000 * banner.getDelaySeconds());
+                } else {
+                    handler.post(() -> showBanner(bannerPopup));
+                }
             } else {
                 long delay = banner.getStartAt().getTime() - now.getTime();
                 handler.postDelayed(() -> showBanner(bannerPopup), delay + (1000 * banner.getDelaySeconds()));
             }
         }
     }
 
-	public void showBannerById(String bannerId) {
-		showBannerById(bannerId, null);
-	}
+    public void showBannerById(String bannerId) {
+        showBannerById(bannerId, null);
+    }
 
     public void showBannerById(String bannerId, String notificationId) {
-    	Log.d("CleverPush/AppBanner", "showBannerById: " + bannerId);
-    	this.getBanners(banners -> {
-			for (Banner banner : banners) {
-				if (banner.getId().equals(bannerId)) {
-					AppBannerPopup popup = new AppBannerPopup(activity, banner);
-
-					if (CleverPush.getInstance(activity).areAppBannersDisabled()) {
-						pendingBanners.add(popup);
-						break;
-					}
-
-					handler.post(() -> showBanner(popup));
-					break;
-				}
-			}
-		}, notificationId);
-	}
+        Log.d("CleverPush/AppBanner", "showBannerById: " + bannerId);
+        this.getBanners(banners -> {
+            for (Banner banner : banners) {
+                if (banner.getId().equals(bannerId)) {
+                    AppBannerPopup popup = new AppBannerPopup(activity, banner);
+
+                    if (CleverPush.getInstance(activity).areAppBannersDisabled()) {
+                        pendingBanners.add(popup);
+                        break;
+                    }
+
+                    handler.post(() -> showBanner(popup));
+                    break;
+                }
+            }
+        }, notificationId);
+    }
 
     private void showBanner(AppBannerPopup bannerPopup) {
+        Date now = new Date();
```

Comment:
If you want the current date use Date.now() to be more explicit about what you're trying to do.

Reply:
We'll rework in SDK 2.0